### PR TITLE
[show-hint addon] Introduced option 'scrollMargin'

### PR DESCRIPTION
**Pull request by nibinqtl**
To show some extra lines before and after the selected line.

### CODE REVIEW

**Comment by marijnh** [comment]
Could you rename this to `scrollMargin` to align it with the [top-level option](https://codemirror.net/doc/manual.html#option_cursorScrollMargin) that does the same in the editor itself?

**Comment by nibinqtl** [comment]
> Could you rename this to `scrollMargin` to align it with the [top-level option](https://codemirror.net/doc/manual.html#option_cursorScrollMargin) that does the same in the editor itself?

Done

File: addon/hint/show-hint.js [hunk 1]
```diff
@@ -379,12 +379,15 @@
     },
 
     scrollToActive: function() {
-      var node = this.hints.childNodes[this.selectedHint]
+      var margin = this.completion.options.scrollMargin || 0;
+      if (! Number.isInteger(margin)) margin = 0;
```
Comment (marijnh) [inline addon/hint/show-hint.js:383]: We are still targeting IE9, which doesn't have `Number.isInteger`. The library as a whole doesn't do run-time type checking, so just dropping this line would be fine.

File: addon/hint/show-hint.js [hunk 2]
```diff
@@ -379,12 +379,15 @@
...
+      var node1 = this.hints.childNodes[Math.max(0, this.selectedHint-margin)];
+      var node2 = this.hints.childNodes[Math.min(this.data.list.length-1, this.selectedHint+margin)];
```
Comment (marijnh) [inline addon/hint/show-hint.js:385]: (Could you follow the whitespace-around-operators style from the surrounding code?)

File: addon/hint/show-hint.js (hunk 1 above)
Comment (nibinqtl) [reply addon/hint/show-hint.js]: done

File: addon/hint/show-hint.js (hunk 2 above)
Comment (nibinqtl) [reply addon/hint/show-hint.js]: done

**Comment by marijnh** [comment]
Thanks!

**Pull request merged**